Code/src/test/java/nl/utwente/group10/haskell/type/UnificationTest.java

```diff
@@ -0,0 +1,59 @@
+package nl.utwente.group10.haskell.type;
+
+import static org.junit.Assert.*;
+import nl.utwente.group10.haskell.catalog.HaskellCatalog;
+import nl.utwente.group10.haskell.exceptions.CatalogException;
+import nl.utwente.group10.haskell.exceptions.HaskellException;
+import nl.utwente.group10.haskell.expr.Apply;
+import nl.utwente.group10.haskell.expr.Expr;
+import nl.utwente.group10.haskell.expr.Ident;
+import nl.utwente.group10.haskell.expr.Value;
+import nl.utwente.group10.haskell.hindley.GenSet;
+
+import org.junit.Test;
+
+public class UnificationTest {
+
+    @Test
+    public void testUnifyUndefined() throws CatalogException, HaskellException {
+        Expr e0 = new Ident("const");
+        Type t0 = e0
+                .analyze(new HaskellCatalog().asEnvironment(), new GenSet())
+                .prune();
+        assertEquals("(a -> (b -> a))", t0.toHaskellType());
+
+        Expr e1 = new Apply(e0, new Ident("undefined"));
+        Type t1 = e1
+                .analyze(new HaskellCatalog().asEnvironment(), new GenSet())
+                .prune();
+        assertEquals("(b -> a)", t1.toHaskellType());
+
+        Expr e2 = new Apply(e1, new Ident("undefined"));
+        Type t2 = e2
+                .analyze(new HaskellCatalog().asEnvironment(), new GenSet())
+                .prune();
+        assertEquals("a", t2.toHaskellType());
```

Comment:
`t2` is hier een `VarT` zonder verdere restricties. Volgens mij is dat correct.

Deze test test dus volgens mij iets anders dan dat je bedoelt. De typechecker mag aan een nieuwe VarT elke naam geven die 'ie wil, en dat is wat hier gebeurt: je kunt er niet van uitgaan dat 'ie 'm `"a"` noemt. Ik heb @jjkester ook gevraagd om naar deze test te kijken, en die zei hetzelfde. Als je nog vragen hebt kun je hem dus even aanschieten.

Voor deze PR zou ik 'm er uithalen.